tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_CcsnCollapse.cpp

```diff
@@ -0,0 +1,250 @@
+      mesh.number_of_grid_points(), 0.0};
+  for (size_t i = 0; i < 3; ++i) {
+    in_coords.get(i) = 1.0e-2 * (log_coords.get(i) + 1.);
+  }
+
+  const size_t num_radial_points = 1;
+  tnsr::I<DataVector, 3, Frame::Inertial> in_coords_large_radius{
+      num_radial_points, 1.0e30};
+
+  INFO("Check Physicality of Results");
+
+  const auto vars = ccsn_progenitor.variables(
+      in_coords,
+      tmpl::list<
+          hydro::Tags::RestMassDensity<DataVector>,
+          hydro::Tags::ElectronFraction<DataVector>,
+          hydro::Tags::SpatialVelocity<DataVector, 3>,
+          hydro::Tags::SpecificEnthalpy<DataVector>,
+          hydro::Tags::Pressure<DataVector>,
+          hydro::Tags::SpecificInternalEnergy<DataVector>,
+          hydro::Tags::LorentzFactor<DataVector>,
+          hydro::Tags::MagneticField<DataVector, 3>,
+          hydro::Tags::DivergenceCleaningField<DataVector>,
+          gr::Tags::Lapse<DataVector>,
+          gr::Tags::Shift<3, Frame::Inertial, DataVector>,
+          gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>,
+          gr::Tags::SqrtDetSpatialMetric<DataVector>,
+          gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DataVector>,
+          gr::Tags::ExtrinsicCurvature<3, Frame::Inertial, DataVector>,
+          ::Tags::deriv<gr::Tags::Lapse<DataVector>, tmpl::size_t<3>,
+                        Frame::Inertial>,
+          ::Tags::deriv<gr::Tags::Shift<3, Frame::Inertial, DataVector>,
+                        tmpl::size_t<3>, Frame::Inertial>,
+          ::Tags::deriv<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>,
+                        tmpl::size_t<3>, Frame::Inertial>>{});
+
+  // Check velocity > c check and provide small interpolation ratio test
+  // coverage
+#ifdef SPECTRE_DEBUG
+  CHECK_THROWS_WITH(
+      ccsn_progenitor_v_grtr_than_c.variables(
+          in_coords, tmpl::list<hydro::Tags::SpatialVelocity<DataVector, 3>>{}),
+      Catch::Matchers::Contains("Spatial velocity"));
+#endif
+
+  // Check radius too large check
+  CHECK_THROWS_WITH(ccsn_progenitor.variables(
+                        in_coords_large_radius,
+                        tmpl::list<hydro::Tags::RestMassDensity<DataVector>>{}),
+                    Catch::Matchers::Contains("Requested radius "));
+
+  // Ensure density is positive
+  const auto& rest_mass_density =
+      get<hydro::Tags::RestMassDensity<DataVector>>(vars);
+
+  CHECK(min(get(rest_mass_density)) > 0.0);
+
+  // Relevant metric variables
+  const auto& spatial_metric =
+      get<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>>(vars);
+  const auto& sqrt_det_spatial_metric_analytic =
+      get<gr::Tags::SqrtDetSpatialMetric<DataVector>>(vars);
+  const auto& det_spatial_metric_numeric =
+      determinant_and_inverse(spatial_metric).first;
+
+  // Metric must always be greater or equal to 1.0 (i.e. flat space)
+  CHECK(abs(min(get(sqrt_det_spatial_metric_analytic)) - 1.0) >= 1.0e-15);
```

Comment:
These "approximate" things can be check with:
```cpp
CHECK(min(get(sqrt_det_spatial_metric_analytic)) approx(1.0));
```
same with the 2 below. We also have `CHECK_ITERABLE_APPROX` that can be applied over `DataVector`s, etc. You can also use custom `approx` variants that have larger scales (e.g. 1.0e-9 instead of 1.0e-14)